articles/introduction-to-knn-algorithm/index.md

```diff
@@ -0,0 +1,232 @@
+
+In simple words, it can be said that we are teaching the machine how to predict or classify by showing some samples (training and testing) of the possible predictions or classifications.
+
+We use train data with features and labels, to enable the machine to learn first. Later, it is being validated or tested using the test data. Thereby, this method of learning is called supervised learning.
+
+### K-Nearest Neighbors algorithm
+K-Nearest Neighbors (KNN) algorithm is one such supervised learning that can be used for classification and regression.
+
+Steps followed in KNN algorithm:
+1. Load the train and test datasets.
+2. Specify or choose the value of `K`.
+3. For each point on the test data perform the following:
+   3.1. Calculate the distance between the point and each point of the training dataset. We can use Euclidean distance or Manhattan distance.
+   3.2. Sort the values in ascending order based on distances.
+   3.3. Find the top K values from the sorted list.
+   3.4. Find the frequency (Mode) of the labels of the top K values.
+   3.5. Assign the mode to the test data point.
+   3.6. The assigned value is the classified or predicted value for the particular test data point.
+
+![KNN algorithm](/engineering-education/introduction-to-knn-algorithm/knn.png)
+*[Image source](https://www.javatpoint.com/k-nearest-neighbor-algorithm-for-machine-learning): KNN algorithm shown visually*
+
+Now, we will learn how to build the above steps from scratch.
+
+### Step by step guide
+Let's learn to build the K-Nearest Neighbors algorithm from scratch. Though, we use certain libraries like `sklearn` containing pre-defined methods to classify or predict using the KNN algorithm; learning to build from scratch would help grasp the concepts better and more effectively.
+
+#### Objective
+For demonstration, we will be making use of a `Diabetes` dataset, which contains various parameters to classify if a person has diabetes or not.
+
+You can download the dataset [here](https://github.com/srishilesh/Machine-learning/blob/master/Classification/KNN/diabetes.csv).
+
+A sample screenshot of the dataset is shown below:
+![Screenshot of the dataset](/engineering-education/introduction-to-knn-algorithm/dataset.PNG)
+*Screenshot of diabetes dataset*
+
+#### Installation
+For building the KNN algorithm from scratch, you can install any version of Python above 3.0.
+
+To download Python, please use [this link](https://www.python.org/downloads/).
+
+On installing Python, you have to install the following libraries:
+- [Numpy](https://numpy.org/install/)
+- [Pandas](https://pandas.pydata.org/docs/getting_started/install.html)
+- [Operator](https://docs.python.org/3/library/operator.html)
+- [Statistics](https://docs.python.org/3/library/statistics.html)
+
```

Comment:
Please include commands to install these libraries, it would be easier for readers to follow:

```bash
pip install numpy
```